src/sql/workbench/contrib/executionPlan/browser/widgets/findExpensiveNodeWidget.ts

```diff
@@ -0,0 +1,311 @@
+				return ExpensiveOperationType.SubtreeCost;
+			case 'actualNumberOfRowsForAllExecutions':
+				return ExpensiveOperationType.ActualNumberOfRowsForAllExecutions;
+			case 'numberOfRowsRead':
+				return ExpensiveOperationType.NumberOfRowsRead;
+			default:
+				return ExpensiveOperationType.Off;
+		}
+	}
+
+	private renderAndStyleWidget(): void {
+		// Expensive Operation Dropdown
+		this._operationNameSelectBoxContainer = DOM.$('expensive-operation-name-select-box .dropdown-container');
+		const operationLabel = DOM.$('expensive-operation-name-select-box-label');
+		operationLabel.innerText = localize('expensiveOperationLabel', 'Metric:');
+
+		this._operationNameSelectBoxContainer.appendChild(operationLabel);
+		this.container.appendChild(this._operationNameSelectBoxContainer);
+
+		this.operationNameSelectBox = new SelectBox([
+			OFF_STRING,
+			ACTUAL_ELAPSED_TIME_STRING,
+			ACTUAL_ELAPSED_CPU_TIME_STRING,
+			COST_STRING,
+			SUBTREE_COST_STRING,
+			ACTUAL_NUMBER_OF_ROWS_FOR_ALL_EXECUTIONS_STRING,
+			NUMBER_OF_ROWS_READ_STRING
+		], COST_STRING, this.contextViewService, this._operationNameSelectBoxContainer);
+
+		this.operationNameSelectBox.render(this._operationNameSelectBoxContainer);
+		this._register(attachSelectBoxStyler(this.operationNameSelectBox, this.themeService));
+
+		this._operationNameSelectBoxContainer.style.width = '200px';
+		this._operationNameSelectBoxContainer.style.marginRight = '5px';
+
+		this._register(this.operationNameSelectBox.onDidSelect(e => {
+			switch (e.selected) {
+				case ACTUAL_ELAPSED_TIME_STRING:
+					this._selectedExpensiveOperationType = ExpensiveOperationType.ActualElapsedTime;
+					break;
+				case ACTUAL_ELAPSED_CPU_TIME_STRING:
+					this._selectedExpensiveOperationType = ExpensiveOperationType.ActualElapsedCpuTime;
+					break;
+				case COST_STRING:
+					this._selectedExpensiveOperationType = ExpensiveOperationType.Cost;
+					break;
+				case SUBTREE_COST_STRING:
+					this._selectedExpensiveOperationType = ExpensiveOperationType.SubtreeCost;
+					break;
+				case ACTUAL_NUMBER_OF_ROWS_FOR_ALL_EXECUTIONS_STRING:
+					this._selectedExpensiveOperationType = ExpensiveOperationType.ActualNumberOfRowsForAllExecutions;
+					break;
+				case NUMBER_OF_ROWS_READ_STRING:
+					this._selectedExpensiveOperationType = ExpensiveOperationType.NumberOfRowsRead;
+					break;
+				default:
+					this._selectedExpensiveOperationType = ExpensiveOperationType.Off;
+			}
+		}));
+
+		// Apply Button
+		const findExpensiveOperationAction = new FindExpensiveOperationAction();
+		this._register(findExpensiveOperationAction);
+
+		const clearExpensiveOperationAction = new TurnOffExpensiveHighlightingOperationAction();
+		this._register(clearExpensiveOperationAction);
+
+		const cancelExpensiveOperationAction = new CancelExpensiveOperationAction();
+		this._register(cancelExpensiveOperationAction);
+
+		const self = this;
+		this.operationNameSelectBox.selectElem.onkeydown = async (ev) => {
```

Comment:
for Enter key press, you shouldn't need to handle it, user is expected to use the apply button to apply changes. for escape key press, it is wrong to do it this way, the correct way is:
1. for mouse user, there should be a close button.
2. for keyboard user, the Escape key handling should happen at the pop out level, with your implementation, if the focus is on the Apply button, and user press Escape, the popup won't be closed.